lib/util/DocumentExtractor.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const extractors = ([
+  {
+    targets: ['delete', 'get'],
+    methods: {
+      extractFromRequest: request => [{ _id: request.input.resource._id }],
+      extractFromResult: request => [request.result],
+      insertInRequest: (documents, request) => {
+        request.input.resource._id = documents[0] && documents[0]._id;
+        return request;
+      },
+      insertInResult: (documents, request) => {
+        request.setResult(documents[0], { status: request.status });
+        return request;
+      }
+    }
+  },
+  {
+    targets: ['mDelete', 'mGet'],
+    methods: {
+      extractFromRequest: request => request.input.body.ids.map(id => ({ _id: id })),
```

Comment:
You should not use `Array.map` for `m*` since its a critical section of code. `for (let i; ... ; i++)` is faster
This remark apply for every `m*` action